pom.xml

```diff
@@ -2139,6 +2139,15 @@
                 <vmtype>org.eclipse.jdt.internal.launching.macosx.MacOSXType</vmtype>
             </properties>
         </profile>
+        <profile>
+            <id>jdk8+</id>
+            <activation>
+                <jdk>[1.8,)</jdk>
+            </activation>
+            <properties>
+                <xDocLint>'-Xdoclint:none'</xDocLint>
```

Comment:
Maybe I am missing something but as in the activation part we use  <jdk>[1.8,)</jdk> it should work with a range from jdk8 to any newer jdk used. That is why I named the profile jdk8+. Always taking account that the newer jdks support the use of the argument -Xdoclint. 

Again maybe I am missing something but as I see in the branch 4.x we set directly the xDocLint property with the value -Xdoclint:none in the properties section. So the property will be always setted at least you pass a different value in the command line. 
That is working in mule 4.x because you are supporting only the use of jdk8 (source=1.8 in the compiler plugin). But in the case of 3.x that we support jdk7 and jdk8 we can't do this because using jdk7 with the argument -Xdoclint will throw an error.